src/fides/api/ops/schemas/saas/saas_config.py

```diff
@@ -203,13 +203,39 @@ def validate_override(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         return values
 
 
+class SaaSRequestMap(BaseModel):
+    """A map of actions to SaaS requests"""
+
+    read: Union[SaaSRequest, List[SaaSRequest]] = []
+    update: Optional[SaaSRequest]
+    delete: Optional[SaaSRequest]
+
+
 class Endpoint(BaseModel):
     """A collection of read/update/delete requests which corresponds to a FidesopsDataset collection (by name)"""
 
     name: str
-    requests: Dict[Literal["read", "update", "delete"], SaaSRequest]
+    requests: SaaSRequestMap
     after: List[FidesCollectionKey] = []
 
+    @validator("requests")
+    def validate_grouped_inputs(
```

Comment:
I couldn't think of a use case where two read requests for the same endpoint would need to have different `grouped_inputs`, we prevent this from happening since that is preferred to having undefined behavior.